src/Checkbox.tsx

```diff
@@ -0,0 +1,74 @@
+import styled from 'styled-components'
+import React, {InputHTMLAttributes, ReactElement, useLayoutEffect, useRef} from 'react'
+import sx, {SxProp} from './sx'
+
+export type CheckboxProps = {
+  /**
+   * Apply indeterminate visual appearance to the checkbox
+   */
+  indeterminate?: boolean
+  /**
+   * Apply inactive visual appearance to the checkbox
+   */
+  disabled?: boolean
+  /**
+   * Forward a ref to the underlying input element
+   */
+  ref?: React.RefObject<HTMLInputElement>
+  /**
+   * Indicates whether the checkbox must be checked
+   */
+  required?: boolean
+
+  /**
+   * Indicates whether the checkbox validation state
+   */
+  validationStatus?: 'error' | 'success' // TODO: hoist to Validation typings
+} & InputHTMLAttributes<HTMLInputElement> &
+  SxProp
+
+const StyledCheckbox = styled.input`
+  cursor: pointer;
+
+  ${props => props.disabled && `cursor: not-allowed;`}
+
+  ${sx}
+`
+
+/**
+ * An accessible, native checkbox component
+ */
+const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
+  (
+    {checked, indeterminate, disabled, sx: sxProp, required, validationStatus, ...rest}: CheckboxProps,
+    ref
+  ): ReactElement => {
+    const checkboxRef = useRef<HTMLInputElement>(null)
```

Comment:
Would it make sense for us to use `useProvedRefOrCreate()` here? https://github.com/primer/react/blob/main/src/hooks/useProvidedRefOrCreate.ts

Reply:
Didn't realise we had this. Awesome. I've swapped it in and checked it works the same 👍